scripts/o3de/tests/ExportScripts/test_export_source_ios_xcode.py

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import pytest
+import pathlib
+from unittest.mock import patch, create_autospec, MagicMock, Mock, PropertyMock
+from o3de.export_project import O3DEScriptExportContext
+
+# Because the export scripts are standalone, we have to manually import them to test
+import o3de.utils as utils
+utils.prepend_to_system_path(pathlib.Path(__file__).parent.parent.parent / 'ExportScripts')
+from export_source_ios_xcode import export_ios_xcode_project
+
+@pytest.mark.parametrize("skip_assets_flag, should_build_tools_flag, expect_get_asset_path, expect_process_command_count", [
```

Comment:
Cool mocking! 

For future are will there be non-mocked tests, or doesn't that eat up way too much time? Then we could check if artifacts really exist, 
Example: does <--output_path>/<project_name>GamePackage/<projectname>.GameLauncher.exe exist

Reply:
i think that'd probably work better in a comprehensive end to end test.